run-dljc.sh

```diff
@@ -100,179 +100,98 @@ fi
-	      USABLE="yes"
-          else
-              # if this last run was under Java 11, try to run
-              # under Java 8 instead
-              if [ "${JAVA_HOME}" = "${JAVA11_HOME}" ]; then
-                  export JAVA_HOME="${JAVA8_HOME}"
-                  echo "couldn't build using Java 11; trying Java 8"
-                  configure_and_exec_dljc
-              else
-                  echo "dljc could not run the build successfully" > ../../../${OUTDIR}-results/${REPO_NAME}-wpi.log
-		  USABLE="no"
-              fi
-          fi
-      fi
-  fi
-  export JAVA_HOME="${JAVA11_HOME}"
-}
+export PATH="${JAVA_HOME}/bin:${PATH}"
+
+mkdir -p "${OUTDIR}"
+mkdir -p "${OUTDIR}-results"
+
+pushd "${OUTDIR}" || exit 5
 
 while IFS='' read -r line || [ "$line" ]
 do    
     REPOHASH=${line}
 
-    REPO=`echo ${REPOHASH} | awk '{print $1}'`
-    HASH=`echo ${REPOHASH} | awk '{print $2}'`
+    REPO=$(echo "${REPOHASH}" | awk '{print $1}')
+    HASH=$(echo "${REPOHASH}" | awk '{print $2}')
 
-    REPO_NAME=`echo ${REPO} | cut -d / -f 5`
+    REPO_NAME=$(echo "${REPO}" | cut -d / -f 5)
 
     # need a layer in the file structure that prevents
     # two repos with the same name from colliding
-    if [ ! -d ${REPO_NAME}-${HASH} ]; then
-        mkdir ${REPO_NAME}-${HASH}
-    fi
+    mkdir -p "${REPO_NAME}-${HASH}"
 
-    pushd ${REPO_NAME}-${HASH}
+    pushd "${REPO_NAME}-${HASH}" || exit 5
     
-    if [ ! -d ${REPO_NAME} ]; then
-        # this environment variable prevents git from prompting for username/password
-        # if the repository no longer exists
-        GIT_TERMINAL_PROMPT=0 git clone ${REPO}
+    if [ ! -d "${REPO_NAME}" ]; then
+        # this environment variable prevents git from prompting for
+	# username/password if the repository no longer exists
+        GIT_TERMINAL_PROMPT=0 git clone "${REPO}"
     else
-        rm -rf ${REPO_NAME}/dljc-out
+        rm -rf "${REPO_NAME}/dljc-out"
     fi
 
     # if the above clone command failed for whatever reason, create an
     # empty directory so that the rest of the commands fail gracefully
     # without messing with the directory structure
-    if [ ! -d ${REPO_NAME} ]; then
-        mkdir ${REPO_NAME}
-    fi
+    mkdir -p "${REPO_NAME}"
 
-    pushd ${REPO_NAME}
+    pushd "${REPO_NAME}" || exit 5
 
-    git checkout ${HASH}
+    git checkout "${HASH}"
 
-    OWNER=`echo ${REPO} | cut -d / -f 4`
+    OWNER=$(echo "${REPO}" | cut -d / -f 4)
 
     if [ "${OWNER}" = "${USER}" ]; then
-        ORIGIN=`echo ${REPOHASH} | awk '{print $3}'`
-        git remote add unannotated ${ORIGIN}
+        ORIGIN=$(echo "${REPOHASH}" | awk '{print $3}')
+        git remote add unannotated "${ORIGIN}"
     fi
+
+    REPO_FULLPATH=$(pwd)
     
-    configure_and_exec_dljc
- 
-    popd 
-    popd
+    popd || exit 5
+
+    RESULT_LOG="${ORIGIN_PWD}/${OUTDIR}-results/${REPO_NAME}-${HASH}-wpi.log"
+    touch "${RESULT_LOG}"
```

Comment:
Why is this `touch` command necessary?

Reply:
The script failed on the line a bit later with the append to the results log if I didn't touch the file first, with a "no such file or directory" message.